test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -456,5 +458,45 @@ public void CheckToOtlpLogRecordExceptionAttributes()
             Assert.Contains(SemanticConventions.AttributeExceptionStacktrace, otlpLogRecordAttributes);
             Assert.Contains(logRecord.Exception.ToInvariantString(), otlpLogRecordAttributes);
         }
+
+        [Fact]
+        public void CheckToOtlpLogRecordRespectsAttributeLimits()
+        {
+            const int attributeCountLimit = 3; // 3 => LogCategory, exception.type and exception.message
+            const int attributeValueMaxLength = 8;
+
+            var logRecords = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.AddInMemoryExporter(logRecords);
+                });
+            });
+
+            var logger = loggerFactory.CreateLogger("OtlpLogExporterTests");
+            logger.LogInformation(new Exception("Exception Message"), "Exception Occurred");
+
+            var logRecord = logRecords[0];
+            var otlpLogRecord = logRecord.ToOtlpLog(attributeCountLimit, attributeValueMaxLength);
+
+            Assert.NotNull(otlpLogRecord);
+            Assert.True(otlpLogRecord.DroppedAttributesCount > 0, "Attributes dropped count is unset");
+
+            var exceptionTypeAtt = TryGetAttribute(otlpLogRecord, SemanticConventions.AttributeExceptionType);
+            Assert.NotNull(exceptionTypeAtt);
+            Assert.Equal(attributeValueMaxLength, exceptionTypeAtt.Value.StringValue.Length);
```

Comment:
Instead of comparing just the length, you could actually compare the content of the strings logged.

Reply:
Done 👍